### PR TITLE
#464 Improve error message when program is missing arm of branching statement

### PR DESCRIPTION
**Pull request by imaqtkatt**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -14,6 +14,7 @@ use diagnostics::{Diagnostics, DiagnosticsConfig, ERR_INDENT_SIZE};
 use net::hvm_to_net::hvm_to_net;
 
 pub mod diagnostics;
+#[allow(clippy::mutable_key_type)]
```
Comment (developedby) [inline src/lib.rs:17]: Why is this needed here in the `mod` declaration?

File: src/lib.rs [hunk 2]
```diff
@@ -197,6 +198,7 @@ pub fn readback_hvm_net(
   let mut diags = Diagnostics::default();
   let net = hvm_to_net(net);
   let mut term = net_to_term(&net, book, labels, linear, &mut diags);
+  #[allow(clippy::mutable_key_type)]
```
Comment (developedby) [inline src/lib.rs:201]: Did this start to flag after a clippy update?

Comment (imaqtkatt) [reply src/lib.rs:201]: I think so, with rust 1.80

File: src/lib.rs (hunk 1 above)
Comment (imaqtkatt) [reply src/lib.rs:17]: To allow this warning inside the `fun` module, the only place we are using HashSet/HashMap

Comment (developedby) [reply src/lib.rs:17]: Shouldn't this be placed in wherever we use the hashmap with mutable key?

Comment (developedby) [reply src/lib.rs:17]: Oh is it because it's too many places? in every single HashMap<Name>

Comment (developedby) [inline src/lib.rs:17]: Add a comment explaining why this is necessary.
Eg. "`Name` is implemented with inner mutability so it triggers this warning, but we know it works."

**Review by developedby — CHANGES_REQUESTED** [review]

File: src/lib.rs (hunk 1 above)
Comment (imaqtkatt) [reply src/lib.rs:17]: Yes

**Review by developedby — APPROVED** [review]
